src/common/Edelstein.Common.Gameplay.Game/Combat/AttackRequest.cs

```diff
@@ -0,0 +1,92 @@
+﻿using Edelstein.Common.Utilities.Packets;
+using Edelstein.Protocol.Gameplay.Game.Combat;
+using Edelstein.Protocol.Utilities.Packets;
+
+namespace Edelstein.Common.Gameplay.Game.Combat;
+
+public class AttackRequest : IAttackRequest, IPacketReadable
+{
+    public AttackRequest(AttackType type) => Type = type;
+
+    public AttackType Type { get; }
+    
+    public int DamagePerMob { get; private set; }
+    public int MobCount { get; private set; }
+    
+    public int SkillID { get; private set; }
+    public int Keydown { get; private set; }
+    
+    public bool IsFinalAfterSlashBlast { get; private set; }
+    public bool IsSoulArrow { get; private set; }
+    public bool IsShadowPartner { get; private set; }
+    public bool IsSerialAttack { get; private set; }
+    public bool IsSpiritJavelin { get; private set; }
+    
+    public bool IsLeft { get; private set; }
+    
+    public int Action { get; private set; }
+    
+    public byte AttackType { get; private set; }
+    
+    public int PartyCount { get; private set; }
+    public int SpeedDegree { get; private set; }
+    
+    public int AttackTime { get; private set; }
+    
+    public int Phase { get; private set; }
+
+    public ICollection<IAttackRequestEntry> Entries { get; } = new List<IAttackRequestEntry>();
+
+    public void ReadFrom(IPacketReader reader)
+    {
+        _ = reader.ReadByte(); // bCurFieldKey
+        _ = reader.ReadInt(); // dr0
+        _ = reader.ReadInt(); // dr1
+
+        var v6 = reader.ReadByte();
+        DamagePerMob = v6 >> 0 & 0xF;
```

Comment:
I understand you don't like the extra parenthesis but for matters of clarity it could be beneficial. Not everyone is fully accustomed to order of precedence for various operator chains.